app/javascript/mastodon/features/compose/components/compose_form.jsx

```diff
@@ -95,6 +103,10 @@ class ComposeForm extends ImmutablePureComponent {
     }
   };
 
+  getMaxCharacters = () => {
+    return this.props.server.getIn(['configuration', 'statuses', 'max_characters']);
```

Comment:
Maybe have a fallback to 500 chars?

```suggestion
    return this.props.server.getIn(['configuration', 'statuses', 'max_characters'], 500);
```